main.js

```diff
@@ -0,0 +1,43 @@
+let books = [];
+
+function getInput() {
+  const book = {};
+  book.title = document.getElementById('bookTitle').value;
+  book.author = document.getElementById('bookAuthor').value;
+  return book;
+}
+
+function removeBook(title) {
+  const book = document.getElementById(title);
+  book.remove();
+  books = books.filter((bookObj) => bookObj.title !== title);
+  localStorage.setItem('array', JSON.stringify(books));
+}
+
+function addBook(bookObj) {
+  const bookList = document.getElementById('book-list');
+  const book = document.createElement('LI');
+  book.setAttribute('id', bookObj.title);
+  book.innerHTML = `<h3> ${bookObj.title} </h3> <h3>${bookObj.author} `;
+  const deleteBtn = document.createElement('button');
+  deleteBtn.innerHTML = 'Delete';
+  deleteBtn.addEventListener('click', () => removeBook(bookObj.title));
+  book.appendChild(deleteBtn);
+  bookList.appendChild(book);
+}
+
+const addButton = document.getElementById('add-btn');
+addButton.addEventListener('click', () => {
+  const book = getInput();
+  books.push(book);
+  localStorage.setItem('array', JSON.stringify(books));
+  addBook(book);
+});
+
+window.onload = () => {
+  books = JSON.parse(localStorage.getItem('array' || '[]'));
+
+  books.forEach((book) => {
+    addBook(book);
+  });
+};
```

Comment:
hey @Emmyn5600  could you help me a little what requirements am i missing? 

I made an array that keeps a collection of books, made a function to add to that, created a function to remove fom the collection and I used the filter method. it is displayed on the top of the page the add button works, the remove button works and the local storage works
thanks :)


.

Reply:
@Emmyn5600 I am not sure what else to do...

Reply:
Hi, @david63011 on my end I am not able to add the book to the book array because of the push keyword which is used to push a new book in the array, please attach the screenshot image that shows what you did so that another reviewer would see your side.

Reply:
Hi @Emmyn5600 maybe I don't understand exactly but for me everything is ok here is a screenshot for you to see or for the next reviewer

https://snipboard.io/vWHpYa.jpg

Reply:
everything is working